Sources/SemanticIndex/UpdateIndexStoreTaskDescription.swift

```diff
@@ -464,3 +466,43 @@ package struct UpdateIndexStoreTaskDescription: IndexTaskDescription {
     }
   }
 }
+
+fileprivate extension Process {
+  /// Run a process with the given arguments. If the number of arguments exceeds the maximum number of arguments allows,
+  /// create a response file and use it to pass the arguments.
+  static func runUsingResponseFileIfTooManyArguments(
+    arguments: [String],
+    workingDirectory: AbsolutePath?,
+    outputRedirection: OutputRedirection = .collect(redirectStderr: false)
+  ) async throws -> ProcessResult {
+    do {
+      return try await Process.run(
+        arguments: arguments,
+        workingDirectory: workingDirectory,
+        outputRedirection: outputRedirection
+      )
+    } catch SystemError.posix_spawn(E2BIG, _) {
```

Comment:
Wouldn't surprise me if this is different on Windows, but I imagine we'll soon see!